crates/compilers/src/compile/resolc/resolc_artifact_output.rs

```diff
@@ -0,0 +1,377 @@
+use alloy_primitives::{hex, Bytes};
+use foundry_compilers_artifacts::{
+    resolc::contract::ResolcContract, BytecodeObject, CompactBytecode, CompactContract,
+    CompactContractBytecode, CompactContractBytecodeCow, CompactDeployedBytecode, DevDoc,
+    SolcLanguage, SourceFile, StorageLayout, UserDoc,
+};
+use foundry_compilers_core::error::SolcIoError;
+use path_slash::PathBufExt;
+use revive_solidity::SolcStandardJsonOutputContractEVM;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    error::Result, resolc::contracts::VersionedContracts, sources::VersionedSourceFiles,
+    ArtifactFile, ArtifactOutput, Artifacts, ArtifactsMap, OutputContext, ProjectPathsConfig,
+};
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Default)]
+pub struct ResolcArtifactOutput();
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct ContractArtifact {
+    /// The contract ABI.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub abi: Option<JsonAbi>,
+    /// The contract metadata.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub metadata: Option<serde_json::Value>,
+    /// The contract developer documentation.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub devdoc: Option<DevDoc>,
+    /// The contract user documentation.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub userdoc: Option<UserDoc>,
+    /// The contract storage layout.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub storage_layout: Option<StorageLayout>,
+    /// Contract's bytecode and related objects
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub evm: Option<SolcStandardJsonOutputContractEVM>,
+    /// The contract optimized IR code.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub ir_optimized: Option<String>,
+    /// The contract PolkaVM bytecode hash.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub hash: Option<String>,
+    /// The contract factory dependencies.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub factory_dependencies: Option<BTreeMap<String, String>>,
+    /// The contract missing libraries.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub missing_libraries: Option<HashSet<String>>,
+}
+
+impl Default for ContractArtifact {
+    fn default() -> Self {
+        Self {
+            abi: None,
+            metadata: None,
+            devdoc: None,
+            userdoc: None,
+            storage_layout: None,
+            evm: None,
+            ir_optimized: None,
+            hash: None,
+            factory_dependencies: None,
+            missing_libraries: None,
+        }
+    }
+}
+
+impl<'a> From<&'a ContractArtifact> for CompactContractBytecodeCow<'a> {
+    fn from(value: &'a ContractArtifact) -> Self {
+        let (standard_abi, compact_bytecode, compact_deployed_bytecode) = create_byte_code(value);
```

Comment:
same as artifacts/resolc/contract.rs: 
- is it intended to set all fields to None if one of the is None? 
- can this be moved to impl block for ContractArtifact?